Packer plugin support UCloud (#7775)
hashicorp/packer: website/source/docs/builders/ucloud-uhost.html.md

---
description: |
    The `ucloud-uhost` Packer builder plugin provides the capability to build
    customized images based on an existing base images.
layout: docs
page_title: 'UCloud Image Builder'
sidebar_current: 'docs-builders-ucloud-uhost'
---

# UCloud Image Builder

Type: `ucloud-uhost`

The `ucloud-uhost` Packer builder plugin provides the capability to build
customized images based on an existing base images.

This builder builds an UCloud image by launching an UHost instance from a source image,
provisioning that running machine, and then creating an image from that machine.

## Configuration Reference

> The following configuration options available for building UCloud images. They are

Reviewer: ```suggestion
The following configuration options are available for building UCloud images. They are
```

segmented below into two categories: required and optional parameters.

In addition to the options listed here, a
[communicator](../templates/communicator.html) can be configured for this
builder.

> \~&gt; **Note:** The bulider doesn't support Windows image for now and only supports CentoOS image and ubuntu image via SSH anthentication with `ssh_user_name` (Required) and `ssh_password` (Optional).  The `ssh_user_name` will be mandatorily set as `root` for CentOS image and `ubuntu` for Ubuntu image by system, the `ssh_password` may cantain 8-30 characters, and at least 2 items of capital letters, lower case letters, numbers and special characters. The special characters include <code>`()~!@#$%^&*-+=_|{}\[]:;'<>,.?/</code>.

Reviewer: ```suggestion
\~&gt; **Note:** The bulider doesn't support Windows image for now and only supports CentoOS and ubuntu images via SSH anthentication with `ssh_username` (Required) and `ssh_password` (Optional).  The `ssh_username` must be `root` for CentOS images and `ubuntu` for Ubuntu images. The `ssh_password` may contain 8-30 characters, and must contain at least 2 capital letters, lower case letters, numbers and special characters. The special characters include <code>`()~!@#$%^&*-+=_|{}\[]:;'<>,.?/</code>.
```

Author: hey @SwampDragons , thanks for the correction on the grammer issues,  however as for the `ssh_password` , what we actually expect to imply is "the `ssh_password` must be consisted with at least 2 items out of the capital letters, lower case letters, numbers and special characters",  let us know if this make sense to you.

Author: We have recommitted about this change, if you have any questions, please point them out.  And i want to know when will this PR can be merged. Thanks again.

> \~&gt; **Note:** The bulider doesn't support Windows images for now and only supports CentoOS and ubuntu images via SSH anthentication with `ssh_username` (Required) and `ssh_password` (Optional).  The `ssh_username` must be `root` for CentOS images and `ubuntu` for Ubuntu images. The `ssh_password` may contain 8-30 characters, and must be consisted with at least 2 items out of the capital letters, lower case letters, numbers and special characters. The special characters include <code>`()~!@#$%^&*-+=_|{}\[]:;'<>,.?/</code>.

Reviewer: Okay, so if it _has to_ have all of the things listed, the most idiomatic way to say that in English is 
"and must consist of at least 2"... instead of "and must be consisted with at least 2"
